app/models.py

```diff
@@ -257,6 +257,44 @@ def get_string_dict():
         }
 
 
+class SPFCheckResult(EnumE):
+    allow = 0
+    fail = 1
+    soft_fail = 1
+    neutral = 2
+    temp_error = 3
+    not_available = 4
+    perm_error = 5
+
+    @staticmethod
+    def get_string_dict():
+        return {
+            "R_SPF_ALLOW": SPFCheckResult.allow,
+            "R_SPF_FAIL": SPFCheckResult.fail,
+            "R_SPF_SOFTFAIL": SPFCheckResult.soft_fail,
+            "R_SPF_NEUTRAL": SPFCheckResult.neutral,
+            "R_SPF_DNSFAIL": SPFCheckResult.temp_error,
+            "R_SPF_NA": SPFCheckResult.not_available,
+            "R_SPF_PERMFAIL": SPFCheckResult.perm_error,
+        }
+
+
+class SpamdResult:
+    def __init__(self):
+        self.dmarc: DmarcCheckResult = DmarcCheckResult.not_available
+        self.spf: SPFCheckResult = SPFCheckResult.not_available
+        self.domain = "unknown"
```

Comment:
We don't set the domain yet right? I wonder if we should distinguish envelope from domain and header from domain as they are used differently in SPF, DKIM, DMARC?

Reply:
We're not using it yet. I'll remove it to make it cleaner for now, but as you say the idea was to add all the relevent info for anti-abuse in this object so we decide what to do based on it. But for now I've just added spf status.